pkg/apis/nmstate/v1alpha1/condition_types.go

```diff
@@ -38,14 +38,14 @@ func NewCondition(conditionType ConditionType, status corev1.ConditionStatus, re
 	return condition
 }
 
-func (conditions *ConditionList) Set(conditionType ConditionType, status corev1.ConditionStatus, reason ConditionReason, message string) {
+func (conditions *ConditionList) Set(conditionType ConditionType, status corev1.ConditionStatus, reason ConditionReason, message string) *ConditionList {
```

Comment:
I feel that returning the list makes the signature odd. The user then needs to ask whether the Set is mutating or not. I see you changed it to make table testing easier, but please, introduce a helper function for tests instead.